src/Acuminator/Acuminator.Utilities/DiagnosticSuppression/SuppressionManager.cs

```diff
@@ -204,14 +209,61 @@ private static SuppressionDiffResult CompareFiles(SuppressionFile oldFile, Suppr
 		{
 			cancellation.ThrowIfCancellationRequested();
 
-			if (settings.SuppressionMechanismEnabled && Instance?.IsSuppressed(semanticModel, diagnostic, cancellation) == true)
+			if (settings.SuppressionMechanismEnabled && 
+			    (Instance?.IsSuppressed(semanticModel, diagnostic, cancellation) == true ||
+			    CheckSuppressedComment(diagnostic, cancellation)))
 			{
 				return;
 			}
 
 			reportDiagnostic(diagnostic);
 		}
 
+		private static bool CheckSuppressedComment(Diagnostic diagnostic, CancellationToken cancellation)
+		{
+			cancellation.ThrowIfCancellationRequested();
+			SyntaxNode root = diagnostic.Location.SourceTree.GetRoot();
+			SyntaxNode node;
+
+			bool containsComment = false;
+
+			// Climb to the hill. Looking for comment on parents nodes.
+			for (node = root.FindNode(diagnostic.Location.SourceSpan); node != root ; node = node.Parent)
```

Comment:
Imho while loop is a bit more readable here. 
`root` and `node` are not checked for null